jwst/photom/photom.py

```diff
@@ -1007,13 +1029,50 @@ def create_2d_conversion(self, model, exptype, conversion, waves, relresps,
         # 2D wavelength grid
         conv_2d = np.interp(wl_array, waves, relresps, left=np.nan, right=np.nan)
 
+        if include_dispersion:
+            dispaxis = get_dispersion_direction(self.exptype, self.grating, self.filter, self.pupil)
+            if dispaxis is not None:
+                dispersion_array = self.get_dispersion_array(wl_array, dispaxis)
+                # Convert dispersion from micron/pixel to angstrom/pixel
+                dispersion_array *= 1.0e4
+                conv_2d /= np.abs(dispersion_array)
+            else:
+                log.warning("Unable to get dispersion direction, so cannot calculate dispersion array")
         # Combine the scalar and 2D conversion factors
         conversion = conversion * conv_2d
         no_cal = np.isnan(conv_2d)
         conversion[no_cal] = 0.
 
         return conversion, no_cal
 
+    def get_dispersion_array(self, wavelength_array, dispaxis):
+        """Create an array of dispersion values from the wavelength array
+
+        Parameters
+        ----------
+        wavelength_array : float
+            2-d array of wavelength values, assumed to be in microns
+        dispaxis : integer
+            Direction of dispersion: 1 = along columns, 2 = along rows
```

Comment:
Minor nit, but I'm not sure I agree with this wording. To me "along columns" says the dispersion varies along a given column, but what's really happening is that it varies *across* columns (i.e. is a function of column number).

Reply:
Yes, I had the directions reversed.  I'll say more clearly that the direction is the direction along which the light is dispersed